conans/client/conf/__init__.py

```diff
@@ -351,6 +351,13 @@ def revisions_enabled(self):
         except ConanException:
             return False
 
+    @property
+    def package_id_mode(self):
+        try:
+            return self.get_item("general.package_id_mode")
+        except ConanException:
+            return None
```

Comment:
Should the value be `None` or semver? Why not present in the `conan.conf`?

Reply:
The values shouldn't contain the "_mode" suffix.

Reply:
I would like to see the possible values as comments in the *conan.conf* too. I even like the idea of having a link to the docs section if we feel this feature is important enough...

Reply:
It is different to not define a new default_package_id_mode, than defining it to semver. This is the logic:

```python
# sha values
        if package_id_mode:
            try:
                getattr(self, package_id_mode)()
            except AttributeError:
                raise ConanException("'%s' is not a known package_id_mode" % package_id_mode)
        else:
            if indirect:
                self.unrelated_mode()
            else:
                self.semver_mode()
```
So if defaults to ``semver`` it will never apply the current ``unrelated_mode()`` that is applying to indirect (transitive, but not declared) dependencies, changing the current behavior.

Maybe we need to discuss this, but I thought that the new default mode would apply to all dependencies, transitive indirect dependencies too.

Reply:
I really fail to see the whole picture now, so I will need a brief explanation